spec/jobs/subscription_confirm_job_spec.rb

```diff
@@ -134,6 +134,48 @@
       expect(job).to receive(:record_order)
     end
 
+    context "when Stripe payments need to be processed" do
+      let(:charge_response_mock) do
+        { status: 200, body: JSON.generate(id: "ch_1234", object: "charge", amount: 2000) }
+      end
+
+      before do
+        allow(order).to receive(:payment_required?) { true }
+        expect(job).to receive(:setup_payment!) { true }
+        stub_request(:post, "https://api.stripe.com/v1/charges")
+          .with(body: /amount/)
+          .to_return(charge_response_mock)
+      end
+
+      context "Stripe SCA" do
+        let(:stripe_sca_payment_method) { create(:stripe_sca_payment_method) }
+        let(:stripe_sca_payment) { create(:payment, amount: 10, payment_method: stripe_sca_payment_method) }
+
+        before do
+          allow(order).to receive(:pending_payments) { [stripe_sca_payment] }
+          expect(stripe_sca_payment_method).to receive(:charge_offline)
+        end
+
+        it "runs the charges in offline mode" do
+          job.send(:confirm_order!, order)
+        end
+      end
+
+      context "Stripe Connect" do
+        let(:stripe_connect_payment_method) { create(:stripe_connect_payment_method) }
+        let(:stripe_connect_payment) { create(:payment, amount: 10, payment_method: stripe_connect_payment_method) }
+
+        before do
+          allow(order).to receive(:pending_payments) { [stripe_connect_payment] }
+          expect(stripe_connect_payment_method).to receive(:charge_offline)
```

Comment:
This doesn't test if the method `charge_offline` actually exists. This test passes regardless of your fix. Your other unit test covers that though. I found that less mocking / more integration testing reveals more bugs and needs fewer changes when refactoring code.

Reply:
Ah, yeah. Good point!

Reply:
See my comment above

Reply:
Definitely agree. I updated the Connect spec 3f39d23dcba7c0b437e0dfdd1290987f826e5dcd and also the SCA spec f7a00dc0d97d4d8a0fb9c27c667051af9a0da9ca